Mathlib/CategoryTheory/Galois/Examples.lean

```diff
@@ -0,0 +1,106 @@
+Authors: Christian Merten
+-/
+import Mathlib.CategoryTheory.Galois.Basic
+import Mathlib.RepresentationTheory.Action.Basic
+import Mathlib.RepresentationTheory.Action.Limits
+import Mathlib.CategoryTheory.Limits.FintypeCat
+import Mathlib.CategoryTheory.Limits.Shapes.Types
+import Mathlib.Logic.Equiv.TransferInstance
+
+/-!
+# Examples of Galois categories and fibre functors
+
+We show that for a group `G` the category of finite `G`-sets is a `PreGaloisCategory` and that the
+forgetful functor to `FintypeCat` is a `FibreFunctor`.
+
+## Todo
+
+* Characterize connected objects in the category of finite `G`-sets as those with transitive
+  `G`-action
+
+-/
+
+universe u v w
+
+namespace CategoryTheory
+
+namespace FintypeCat
+
+open Limits Functor PreGaloisCategory
+
+private noncomputable def imageComplement {X Y : FintypeCat.{u}} (f : X ⟶ Y) :
+    FintypeCat.{u} := by
+  haveI : Fintype (↑(Set.range f)ᶜ) := Fintype.ofFinite _
+  exact FintypeCat.of (↑(Set.range f)ᶜ)
+
+private def imageComplementIncl {X Y : FintypeCat.{u}}
+    (f : X ⟶ Y) : imageComplement f ⟶ Y :=
+  Subtype.val
+
+variable (G : Type u) [Group G]
+
+/- Given `f : X ⟶ Y` for `X Y : Action FintypeCat (MonCat.of G)`, the complement of the image
+of `f` has a natural `G`-action. -/
+private noncomputable def Action.imageComplement {X Y : Action FintypeCat (MonCat.of G)}
+    (f : X ⟶ Y) : Action FintypeCat (MonCat.of G) where
+  V := FintypeCat.imageComplement f.hom
+  ρ := MonCat.ofHom <| {
+    toFun := fun g y ↦ Subtype.mk (Y.ρ g y.val) <| by
+      intro ⟨x, h⟩
+      apply y.property
+      use X.ρ g⁻¹ x
+      calc (X.ρ g⁻¹ ≫ f.hom) x
+          = (Y.ρ g⁻¹ * Y.ρ g) y.val := by rw [f.comm, FintypeCat.comp_apply, h]; rfl
+        _ = y.val := by rw [← map_mul, mul_left_inv, Action.ρ_one, FintypeCat.id_apply]
+    map_one' := by simp only [Action.ρ_one]; rfl
+    map_mul' := fun g h ↦ FintypeCat.hom_ext _ _ <| fun y ↦ Subtype.ext <| by
+      exact congrFun (MonoidHom.map_mul Y.ρ g h) y.val
+  }
+
+private def Action.imageComplementIncl {X Y : Action FintypeCat (MonCat.of G)} (f : X ⟶ Y) :
+    Action.imageComplement G f ⟶ Y where
+  hom := FintypeCat.imageComplementIncl f.hom
+  comm _ := rfl
+
+private instance {X Y : Action FintypeCat (MonCat.of G)} (f : X ⟶ Y) :
+    Mono (Action.imageComplementIncl G f) := by
+  apply Functor.mono_of_mono_map (forget _)
+  apply ConcreteCategory.mono_of_injective
+  exact Subtype.val_injective
+
+attribute [-instance] Fin.instMulFin
+attribute [-instance] Distrib.toMul
+
+/-- The category of finite sets has quotients by finite groups in arbitrary universes. -/
+instance (G : Type u) [Group G] [Finite G] : HasColimitsOfShape (SingleObj G) FintypeCat.{w} := by
+  obtain ⟨n, ⟨e⟩⟩ := Finite.exists_equiv_fin G
+  letI groupH : Group (Fin n) := Equiv.group e.symm
+  letI e' : (Fin n) ≃* G := Equiv.mulEquiv e.symm
+  letI eq : SingleObj G ≌ SingleObj (Fin n) := e'.symm.toSingleObjEquiv
+  exact Limits.hasColimitsOfShape_of_equivalence eq.symm
```

Comment:
From this, it would be nice to extract a lemma `Finite.exists_mulEquiv` which would show the existence of a `mulEquiv` `G ≃* G'` to a (finite) group `G'` in `Type`, and then use this lemma here.

Reply:
See the latest commit, I moved it to `Mathlib.Logic.Equiv.TransferInstance`, thats what `#find_home` printed. I did not know how to name it, feel free to suggest something else.

Reply:
Maybe `Finite.exists_type_zero_nonempty_mulEquiv`?

Reply:
I moved this part into a new `namespace Finite` block with your suggested name.